src/mono/mono/utils/mono-threads-posix.c

```diff
@@ -306,6 +304,9 @@ mono_memory_barrier_process_wide (void)
 	status = mono_mprotect (memory_barrier_process_wide_helper_page, mono_pagesize (), MONO_MMAP_NONE);
 	g_assert (status == 0);
 
+	status = mono_mprotect (memory_barrier_process_wide_helper_page, mono_pagesize (), MONO_MMAP_READ | MONO_MMAP_WRITE);
```

Comment:
Can we add comment describing the significance of the ordering here?

Reply:
added a comment summarizing why we reset the protections.